tests/web/src/main.rs

```diff
@@ -0,0 +1,95 @@
+use winit::{
+    event::{
+        device::{GamepadEvent, GamepadHandle},
+        Event, WindowEvent,
+    },
+    event_loop::{ControlFlow, EventLoop},
+    window::WindowBuilder,
+};
+
+// wasm-pack test --firefox --headless
+
+// use wasm_bindgen::prelude::*;
+
+// #[wasm_bindgen]
+// extern "C" {
+//     // Use `js_namespace` here to bind `console.log(..)` instead of just
+//     // `log(..)`
+//     #[wasm_bindgen(js_namespace = console)]
+//     fn log(s: &str);
+
+//     // The `console.log` is quite polymorphic, so we can bind it with multiple
+//     // signatures. Note that we need to use `js_name` to ensure we always call
+//     // `log` in JS.
+//     #[wasm_bindgen(js_namespace = console, js_name = log)]
+//     fn log_u32(a: u32);
+
+//     // Multiple arguments too!
+//     #[wasm_bindgen(js_namespace = console, js_name = log)]
+//     fn log_many(a: &str, b: &str);
+// }
+
+// macro_rules! console_log {
+//   // Note that this is using the `log` function imported above during
+//   // `bare_bones`
+//   ($($t:tt)*) => (log(&format_args!($($t)*).to_string()))
+// }
+
+extern crate wasm_bindgen_test;
+use wasm_bindgen_test::*;
+
+wasm_bindgen_test_configure!(run_in_browser);
+
+#[wasm_bindgen_test]
```

Comment:
I'm not sure automated testing fits here, because very little is actually tested here. All we observe is that the application starts and runs without error, but not much else.

Reply:
Which is weird in this test is that the CloseRequested event is immediatly triggered.
How did you test your implementation?

Reply:
I ran the implementation in the browser, mostly testing the stdweb side via `cargo-web` during development. For the web-sys side, the testing process is a little more onerous:

- Compile one of the examples
- Create an HTML page that embeds it
- Run a webserver locally that can reach the HTML file and the relevant JS and wasm files
- Open the page in the browser and make sure the events are generated correctly

Reply:
Ok sorry, noob things, I figured out the problem. I needed `console_error_panic_hook` dependency to stack trace errors. I found ones. Everything fixed, I can continue the implementation =).